### PR TITLE
Fix issue with symlinks in directories for nfpm

### PR DESCRIPTION
**Pull request by Felixoid**
It is the first step necessary to solve https://github.com/goreleaser/nfpm/issues/428

### CODE REVIEW

**Comment by Felixoid** [comment]
I am sorry, @caarlos0, are checks working even? They haven't started for https://github.com/goreleaser/fileglob/commit/f338501c4ca44e8d5fd263f0ebdd92160f0a7a03 in ~8 hours

**Comment by caarlos0** [comment]
> I am sorry, @caarlos0, are checks working even? They haven't started for [f338501](https://github.com/goreleaser/fileglob/commit/f338501c4ca44e8d5fd263f0ebdd92160f0a7a03) in ~8 hours

it asks for a maintainer to approve, just did

File: glob_test.go [hunk 1]
```diff
@@ -450,6 +450,25 @@ func TestGlob(t *testing.T) { // nolint:funlen
...
+		fsPath := testFS.Path()
+		os.Symlink("file", filepath.Join(fsPath, "./a/working-symlink"))
+		os.Symlink("non-existent", filepath.Join(fsPath, "./a/broken-symlink"))
+
+		matches, err := Glob("a/*", MatchDirectoryIncludesContents, WithFs(testFS))
+		require.NoError(t, err)
+		require.Equal(t, []string{
+			"a/broken-symlink",
+			"a/file",
+			"a/working-symlink",
```
Comment (caarlos0) [inline glob_test.go:469]: if I understand it correctly, without the patch, fileglob would follow the symlink and this entry wouldn't be here, as `a/file` is already, correct?

Comment (Felixoid) [reply glob_test.go]: Please, take another look. The fix is for patterns of symlinks, despite if they are valid of point to non-existent files

**Comment by Felixoid** [comment]
I don't see what's the issue with linter, actually

**Comment by Felixoid** [comment]
Does it look good now?

**Comment by caarlos0** [comment]
thanks!

**Pull request merged**